repos/keg-components/src/components/form/checkGroup/checkGroup.js

```diff
@@ -0,0 +1,115 @@
+import React, { useCallback, useState, useMemo } from 'react'
+import { View } from 'KegView'
+import { Text } from '../../typography/text'
+import { Checkbox } from 'KegCheckbox'
+import { useStylesMemo } from '@keg-hub/re-theme'
+import { isFunc } from '@keg-hub/jsutils'
+
+/**
+ * Simple header for CheckGroup, without a checkbox
+ * @param {Object} props
+ * @param {string} props.title - title of header
+ * @param {Object} props.style - style rules to apply to the Text element
+ * @param {*} props.* - remaining props are passed to the Text element
+ */
+export const SimpleHeader = ({ title, style, ...rest }) => {
+  const textStyle = useMemo(() => ({ ...style, marginLeft: 27 }), [ style ])
+  return (
+    <Text 
+      style={textStyle} 
+      { ...rest }
+    >
+      { title }
+    </Text>
+  )
+}
+
+/**
+ * Header for CheckGroup, with a checkbox for select-all functionality
+ * @param {Object} props
+ * @param {string} props.title - title of header
+ * @param {Object} props.style - style rules to apply to the header
+ * @param {Function?} props.onPress - handler for checkbox press
+ * @param {boolean} props.checked - initial value of header checkbox
+ */
+export const CheckboxHeader = ({ title, style, onPress, checked }) => {
+  const headerStyles = useMemo(() => ({
+    main: style,
+    content: {
+      right: {}
+    }
+  }), [ style ])
+
+  const onChangeHandler = useCallback((_, val) => onPress?.(val), [onPress])
+
+  return ( 
+    <Checkbox 
+      RightComponent={title}
+      onChange={onPress}
+      styles={headerStyles}
+      checked={checked}
+      onChange={onChangeHandler}
+      close
+    />
+  )
+}
+
+/**
+ * A group of checkbox items with a header.
+ * Will include a select-all checkbox in the header IF you define the children as a function (@see story examples)
+ * @param {Object} props
+ * @param {string} props.className - css class name
+ * @param {string} props.title - title of the group
+ * @param {boolean?} props.initChecked - initial checked value of the group header box, if you are using it
+ * @param {Function?} props.onGroupPress - handler of header checkbox
+ * @param {*} props.children - components in the group (@see CheckboxGroup.Item)
+ * @param {Object?} props.styles
+ * @param {Object?} props.styles.main
+ * @param {Object?} props.styles.header
+ */
+export const CheckGroup = ({ className, title, children, styles, initChecked=false, onGroupPress }) => {
+  const groupStyles = useStylesMemo('form.checkGroup', styles)
+
+  // if children is defined as a function, it's assumed it needs the groupProps for select-all behavior
+  const useCheckboxHeader = isFunc(children)
```

Comment:
The name makes me think this is a `hook`. I hate that this is the case because I like the convention, but React kind of stole it with hooks.
Maybe update it to `renderCheckboxHeader` || `applyCheckboxHeader`

Reply:
good point. updated